src/ServiceControl/EventLog/AuditEventLogWriter.cs

```diff
@@ -3,29 +3,30 @@
     using Contracts.EventLog;
     using NServiceBus;
     using Raven.Client;
+    using ServiceControl.Infrastructure.DomainEvents;
     using ServiceControl.Infrastructure.SignalR;
 
     /// <summary>
     /// Only for events that have been defined (under EventLog\Definitions), a logentry item will
     /// be saved in Raven and an event will be raised.
     /// </summary>
-    public class GenericAuditHandler : IHandleMessages<IEvent>
+    public class AuditEventLogWriter : IDomainHandler<IDomainEvent>
     {
         static string[] emptyArray = new string[0];
         private readonly IBus bus;
         private readonly GlobalEventHandler broadcaster;
         private readonly IDocumentStore store;
         private readonly EventLogMappings mappings;
 
-        public GenericAuditHandler(IBus bus, GlobalEventHandler broadcaster, IDocumentStore store, EventLogMappings mappings)
+        public AuditEventLogWriter(IBus bus, GlobalEventHandler broadcaster, IDocumentStore store, EventLogMappings mappings)
         {
             this.bus = bus;
             this.broadcaster = broadcaster;
             this.store = store;
             this.mappings = mappings;
         }
 
-        public void Handle(IEvent message)
+        public void Handle(IDomainEvent message)
```

Comment:
https://github.com/Particular/ServiceControl/blob/9af43ddc2358eee3b74f528f8294c3141e886cdf/src/ServiceControl/EventLog/AuditEventLogWriter.cs#L36 feels leaky now since this is no longer a handler. I guess we need to remove the bus instance and somehow float the message id in over the domain event

Reply:
I am going to switch to using guids for ID of event log events